store/src/java/com/zimbra/cs/mailbox/ContactBackupThread.java

```diff
@@ -148,99 +100,71 @@ private void setContactBackupLastMailboxId(int id) {
-                        if (folder != null) {
-                            createBackup(octxt, mbox, account, folder, startTime);
-                            purgeOldBackups(octxt, mbox, folder, startTime);
-                        } else {
-                            success = false;
-                            ZimbraLog.contactbackup.info("contact backup folder not found, continuing to next mailbox");
-                        }
-                        // set current mailbox id as last processed mailbox
-                        if (success) {
-                            setContactBackupLastMailboxId(mbox.getId());
-                        }
+            ZimbraLog.contactbackup.debug("starting to work with mailbox %d", mailboxId);
+            ZimbraLog.addMboxToContext(mailboxId);
+            success = true;
+            try {
+                Mailbox mbox = MailboxManager.getInstance().getMailboxById(mailboxId);
+                sb.append(mailboxId).append(",");
+                Account account = mbox.getAccount();
+                sb.append(account.getMail()).append(",");
+                ZimbraLog.addAccountNameToContext(account.getName());
+                if (account.isFeatureContactBackupEnabled() && !account.isIsSystemAccount() && !account.isIsSystemResource() && account.isAccountStatusActive()) {
+                    OperationContext octxt = new OperationContext(account);
+                    Folder folder = getContactBackupFolder(octxt, mbox, true);
+                    if (folder != null) {
+                        createBackup(octxt, mbox, account, folder, startTime);
+                        purgeOldBackups(octxt, mbox, folder, startTime);
                     } else {
-                        ZimbraLog.contactbackup.debug("contact backup skipped: feature is disabled/account is inactive/it's a system account");
+                        success = false;
+                        ZimbraLog.contactbackup.info("contact backup folder not found for %d, continuing to next mailbox", mailboxId);
+                    }
+                    // set current mailbox id as last processed mailbox
+                    if (success) {
+                        setContactBackupLastMailboxId(mailboxId);
+                        sb.append("success");
                     }
-                } catch (Exception e) {
-                    ZimbraLog.contactbackup.warn("backup/purge failed for mailbox %d, continuing to next mailbox", mailboxId, e);
+                } else {
+                    ZimbraLog.contactbackup.debug("contact backup skipped for %d: feature is disabled/account is inactive/it's a system account", mailboxId);
+                    sb.append("skipped");
                 }
-                ZimbraLog.clearContext();
+            } catch (Exception e) {
+                ZimbraLog.contactbackup.warn("backup/purge failed for mailbox %d, continuing to next mailbox", mailboxId, e);
```

Comment:
Please move all stack traces to debug level

Reply:
done